org.adempiere.base/src/org/compiere/model/MPInstance.java

```diff
@@ -630,4 +630,18 @@ public static class PInstanceInfo {
 		public boolean isDirectPrint = false;
 		public int estimate;
 	}
+	
+	/**
+	 * 	Before Save
+	 *	@param newRecord new
+	 *	@return true
+	 */
+	protected boolean beforeSave (boolean newRecord)
+	{
+		if (newRecord) {
+			setAD_Session_ID(Env.getContextAsInt(Env.getCtx(), "#AD_Session_ID"));

```

Comment:
should use Env.AD_Session_ID constant instead. and perhaps we should check Env.getContextAsInt(Env.getCtx(), Env.AD_Session_ID)) is > 0 here.